test/libsolidity/syntaxTests/experimental/inference/polymorphic_function_call_via_variable.sol

```diff
@@ -0,0 +1,20 @@
+pragma experimental solidity;
+
+type T;
+type U(A);
+
+function f(x, y: X, z: U(Y)) {}
+
+function run(a: T, b: U(T), c: U(U(T))) {
+    let fRef = f;
+
+    // FIXME: The second call does not unify. Apparently the call via a function pointer does not
+    // get fresh type variables like a direct call.
+    fRef(a, a, b);
+    fRef(b, b, c);
```

Comment:
This currently does not work. Variables in `fRef` won't unify with both calls. Perhaps this was missing a `polymorphicInstance()` in the implementation?

I can address it with a separate PR after I'm done with the current work on fixed type variables. Or maybe I should just start creating issues for these things?

Reply:
This *should not* unify. This is exactly what it means to *not* have let-polymorphism.

Reply:
There is no way to fix this other than *introducing* let-polymorphism in the first place, which would complicate the entire mechanism significantly :-).

Reply:
ok then. I'll remove the FIXME :)

Reply:
Done.